### PR TITLE
CW bugfix for #1754

### PR DESCRIPTION
**Pull request by m-chichikalov**
Found couple bugs by adding unit tests for this cw state machine. More comming.
I'm prety sure the Iambic_B not working properly and Ultematic mode as well. 
The tests [here](https://github.com/m-chichikalov/uhsdr-unit-tests/blob/56ae6575f5a69bdc460488178e29ff550ae7cc34/tests/tes_cw_gen_host.cpp#L216).

### CODE REVIEW

**Comment by df8oe** [comment]
Is line 243 of the tests intended as you want - or is it a typo? At line 259 there is following something very similar - but not identical. If this is intended: please can you rename line 243 so that it is not confusing for a typo?

File: mchf-eclipse/drivers/audio/cw/cw_gen.h [hunk 1]
```diff
@@ -16,15 +16,14 @@
...
 void    CwGen_Init();
 
 void    CwGen_PrepareTx();
 void    CwGen_SetSpeed();
 
-bool    CwGen_TimersActive();
 bool    CwGen_Process( float32_t *i_buffer, float32_t *q_buffer, uint32_t size );
 
+// There is no real needs to call these functions. Going remove after adding more tests.
 void    CwGen_DahIRQ();
```
Comment (db4ple) [inline mchf-eclipse/drivers/audio/cw/cw_gen.h:28]: What do you mean with this statement? These functions are called from the interrupt handler for the PTT/paddle GPIOs, see https://github.com/df8oe/UHSDR/blob/def4f73f1af6d081b2fb43cffb458f312f4fa6ed/mchf-eclipse/src/uhsdr_main.c#L62.

**Review by db4ple — CHANGES_REQUESTED** [review]
Please follow the style guideline for code consistently: https://github.com/df8oe/UHSDR/blob/active-devel/CONTRIBUTING.md#coding-style-guide

Thank you!

**Comment by db4ple** [comment]
BTW, interesting setup for testing. How can I setup this to build and run on my machine? 

EDIT: After looking into it, I can see that you are not running on target, only on the host for the moment. Got it.

**Comment by m-chichikalov** [comment]
> Is line 243 of the tests intended as you want - or is it a typo? At line 259 there is following something very similar - but not identical. If this is intended: please can you rename line 243 so that it is not confusing for a typo?

Hi, thank you for review. Both of these test has similar "when" condition but they test different behavior. Everything fine there.

File: mchf-eclipse/drivers/audio/cw/cw_gen.h (hunk 1 above)
Comment (m-chichikalov) [reply mchf-eclipse/drivers/audio/cw/cw_gen.h:28]: > What do you mean with this statement? These functions are called from the interrupt handler for the PTT/paddle GPIOs, see
> 
> https://github.com/df8oe/UHSDR/blob/def4f73f1af6d081b2fb43cffb458f312f4fa6ed/mchf-eclipse/src/uhsdr_main.c#L62
> 
> .

I know, but answer why and for what exactly?

**Comment by df8oe** [comment]
"Eclipse autoformat" is not what we want because it does not remove the newly added format issues but produces noise which makes it more difficult to identify code changes. Please revert the autoformat changes (they add some new format issues indeed) and remove the format issues following our  style guideline for code consistently manually at your added code.

**Pull request merged**